rmf_site_editor/src/site/drawing.rs

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+use crate::{
+    interaction::Selectable,
+    shapes::make_flat_rectangle_mesh,
+    site::{Category, CurrentSite, DefaultFile},
+};
+use bevy::prelude::*;
+use bevy::utils::HashMap;
+use rmf_site_format::{AssetSource, DrawingMarker, PixelsPerMeter, Pose};
+
+use std::path::PathBuf;
+
+// We need to keep track of the drawing data until the image is loaded
+// since we will need to scale the mesh according to the size of the image
+#[derive(Default)]
+pub struct LoadingDrawings(pub HashMap<Handle<Image>, (Entity, Pose, PixelsPerMeter)>);
+
+fn get_current_site_path(
+    current_site: Res<CurrentSite>,
+    site_files: Query<(Entity, &DefaultFile)>,
+) -> Option<PathBuf> {
+    let site_entity = (*current_site).0.unwrap();
+    let site_file = site_files.iter().find(|&el| el.0 == site_entity);
```

Comment:
In fact this whole function can be reduced to

```
fn get_current_site_path(
    current_site: Res<CurrentSite>,
    site_files: Query<&DefaultFile>,
) -> Option<PathBuf> {
    let site_entity = (*current_site).0?;
    site_files.get(site_entity).map(|f| f.0.clone()).ok()
}
```

Let me know if you have any questions about how/why that works. You'll need to tweak the other functions in this module for the change in `site_files`'s type.